File: model/src/seed/SeedReader.java
package seed;

import data.DataReader;

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;
import java.util.StringTokenizer;

public class SeedReader implements DataReader<Seed> {

    private static final String fileName = "resources\\seeds_dataset.txt";

    @Override
    public List<Seed> readData() throws IOException {
        List<Seed> seeds = new ArrayList<>();
        try (BufferedReader input = new BufferedReader(new FileReader(fileName))) {
            String nextLine;
            while ((nextLine = input.readLine()) != null) {
                StringTokenizer tokenizer = new StringTokenizer(nextLine);
                seeds.add(readSeed(tokenizer));
            }
        }
        return seeds;
    }

    private Seed readSeed(StringTokenizer tokenizer) {
        double area = Double.parseDouble(tokenizer.nextToken());
        double perimeter = Double.parseDouble(tokenizer.nextToken());
        double compactness = Double.parseDouble(tokenizer.nextToken());
        double kernelLength = Double.parseDouble(tokenizer.nextToken());
        double kernelWidth = Double.parseDouble(tokenizer.nextToken());
        double asymmetryCoefficient = Double.parseDouble(tokenizer.nextToken());
        double kernelGrooveLength = Double.parseDouble(tokenizer.nextToken());
        int classId = Integer.parseInt(tokenizer.nextToken());

        return new Seed(
                area,
                perimeter,
                compactness,
                kernelLength,
                kernelWidth,
                asymmetryCoefficient,
                kernelGrooveLength,
                classId);
    }
}

File: model/src/seed/SeedReader.kt
package seed

import data.DataReader
import java.io.BufferedReader
import java.io.FileReader
import java.io.IOException
import java.util.*

class SeedReader : DataReader<Seed> {

    @Throws(IOException::class)
    override fun readData(): List<Seed> {
        val seeds = ArrayList<Seed>()
        BufferedReader(FileReader(fileName)).use { input ->
            for (nextLine in input.lines()) {
                val tokenizer = StringTokenizer(nextLine)
                seeds.add(readSeed(tokenizer))
            }
        }
        return seeds.filter { seed -> Math.abs(seed.result) > 0.1  }
    }

    private fun readSeed(tokenizer: StringTokenizer): Seed {
        val area = java.lang.Double.parseDouble(tokenizer.nextToken())
        val perimeter = java.lang.Double.parseDouble(tokenizer.nextToken())
        val compactness = java.lang.Double.parseDouble(tokenizer.nextToken())
        val kernelLength = java.lang.Double.parseDouble(tokenizer.nextToken())
        val kernelWidth = java.lang.Double.parseDouble(tokenizer.nextToken())
        val asymmetryCoefficient = java.lang.Double.parseDouble(tokenizer.nextToken())
        val kernelGrooveLength = java.lang.Double.parseDouble(tokenizer.nextToken())
        val classId = when (Integer.parseInt(tokenizer.nextToken())) {
            1 -> 1
            2 -> 0
            else -> -1
        }

        return Seed(
                area,
                perimeter,
                compactness,
                kernelLength,
                kernelWidth,
                asymmetryCoefficient,
                kernelGrooveLength,
                classId)
    }

    companion object {

        private val fileName = "resources\\seeds_dataset.txt"
    }
}
